### ISSUE TITLE
Block scoped declarations (let, const, function,class) not yet supported outside strict mode

### ISSUE THREAD

**Issue by crazyabdul** [yagop/node-telegram-bot-api#275]
> Bug Report

I have read:

* [Usage information](https://github.com/yagop/node-telegram-bot-api/tree/master/doc/usage.md)
* [Help information](https://github.com/yagop/node-telegram-bot-api/tree/master/doc/help.md)

I am using the latest version of the library.
v 0.26.0

### Expected Behavior

I was trying to run the bot after writing some lines of code. I expected that after running`node index.js`, 
there will be no errors and when i send a request from telegram, get the desired result.
### Actual Behavior

what happened instead was this error in the console

~~~
class TelegramBot extends EventEmitter {
^^^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (E:\Code\dobo\node_modules\node-telegram-bot-api\index.js:12:20)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
~~~

So i run this instead `node --use_strict index.js` and got different set of errors

~~~
constructor(token, options = {}) {
                             ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (E:\Code\dobo\node_modules\node-telegram-bot-api\index.js:12:20)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
~~~

So after a while, I went through `telegram.js` and `telegramPolling.js` and changed the following function parameters
1. options = {} to options
2. forms = {} to forms
3. fileOpts = {} to fileOpts

and in some cases, initialized the variable to an empty object within the function. For example
~~~
startPolling(options) {
    options = {};
    if (this.hasOpenWebHook()) {
      return Promise.reject(new Error('Polling and WebHook are mutually exclusive'));
    }
    options.restart = typeof options.restart === 'undefined' ? true : options.restart;
    if (!this._polling) {
      this._polling = new TelegramBotPolling(this._request.bind(this), this.options.polling, this.processUpdate.bind(this));
    }
    return this._polling.start(options);
  }
~~~

It was after taking these steps that it finally worked

### Steps to reproduce the Behavior

1. write an index.js file with following code
~~~
var token = '<insert token here>';

var Bot = require('node-telegram-bot-api'),
    bot = new Bot(token, { polling: true });

bot.onText(/^\/say_hello (.+)$/, function (msg, match) {
  var name = match[1];
  bot.sendMessage(msg.chat.id, 'Hello ' + name + '!').then(function () {
    // reply sent!
  });
});
~~~

2. run `node index.js`

> Question

My project, as at now, only tests sending messages but I could test other features  If given approval

**Comment by preco21** [comment]
What `node` version are you currently using?

**Comment by crazyabdul** [comment]
@preco21 node version 4.4.3

**Comment by preco21** [comment]
I guess some of features are not supported in `node@4`: https://kangax.github.io/compat-table/es6/#node4

Could you upgrade your node to 6 or 7?

**Comment by yagop** [comment]
Would be better to target a 4.x Node.js version.
For some reason, Travis tests for version 4.x are passing.
What do you think @GochoMugo ?

**Comment by preco21** [comment]
https://github.com/yagop/node-telegram-bot-api/blob/master/index.js

Seems `index.js` is trying to use fallback code when version of node is 0.x.

Still have no idea for about TravisCI..

**Comment by preco21** [comment]
Maybe this [opts](https://github.com/yagop/node-telegram-bot-api/blob/master/test/mocha.opts) file is causing the problem? It is requiring `babel-register` internally, so I guess this is reason how it could worked even on node v4.

**Comment by GochoMugo** [comment]
@preco21 Thanks for looking into this issue. It seems Node v4 does **not** support the syntax fully.

We can have the fallback work also for Node v4.x. However, I was hoping to only deprecate Node v0.x! Do we deprecate v4.x too?

Also, we should check if it runs on v5.x, v6.x, v7.x!

**Comment by crazyabdul** [comment]
@preco21 It seems the quickest solution for me is to upgrade node to version 6. I'll update this thread on how it turns within the next few hours.

**Comment by GochoMugo** [comment]
If you can use the latest version of Node, I would suggest you go with that. However, since we are promising our users that we support Node v4.x, we need to fix this.

**Comment by crazyabdul** [comment]
@gochomugo I could look into adding support for node v4 upwards

**Comment by GochoMugo** [comment]
@crazyabdul We could really do with some help! Thanks in advance!

**Comment by GochoMugo** [comment]
@crazyabdul But we should discuss how to achieve that, to avoid having you do so much work only to be discarded!

**Comment by preco21** [comment]
What about just simply do:

```javascript
const majorVersion = process.versions.node.split('.')[0];
if (majorVersion === '0') {
  const deprecate = require('depd')('node-telegram-bot-api');
  deprecate('Node.js v0.12 and below will no longer be supported in the future');

  module.exports = require('./lib/telegram');
} else if (majorVersion === '4') {
  module.exports = require('./lib/telegram');
} else {
  module.exports = require('./src/telegram');
}
```

Also, you may need to fix `mocha` config to correct TravisCI.

**Comment by crazyabdul** [comment]
@gochoMugo you earlier mentioned something about a fallback. What versions of node are those fallbacks for and how were they implemented. I'd want to learn from what the team has done so far before coming with a plan

**Comment by preco21** [comment]
I guess you guys could consider about using [babel-preset-env](https://github.com/babel/babel-preset-env) over static `es2015` preset.

**Comment by crazyabdul** [comment]
@preco21 then i can put the 2 js files I made node v4 compatible in `./src/telegram`  
I think that'll solve the issue

**Comment by preco21** [comment]
@crazyabdul I guess it should be on `index.js` but not on `src/`. It does not need to be transpiled.

**Comment by crazyabdul** [comment]
@preco21yeah. I see that now
@gochoMugo with @preco21 code snippet and the changes i mentioned in the bug report, this can be resolved fairly quickly.
What do you think?

**Comment by GochoMugo** [comment]
My major issue is whether we should deprecate node v4.x, or fix the syntax to support node v4.x ? We really want to move away from babel!

**Comment by crazyabdul** [comment]
@gochoMugo I came across this [pdf](https://nodejs.org/static/documents/2016-survey-report.pdf) showing stats on the various node versions

On page 18, it shows that majority of people use v4 so it might be worth fixing the syntax

**Comment by crazyabdul** [comment]
@gochoMugo what alternative to babel are you considering?

**Comment by GochoMugo** [comment]
@crazyabdul we are hoping to use the source as is i.e. **no** transpiling!!!

**Comment by yagop** [comment]
According with https://nodesource.com/node-by-numbers v6 is more popular

**Comment by yagop** [comment]
But supporting v4 without transpiling would be the best IMO

**Comment by crazyabdul** [comment]
@yagop checked your link out and it is clearly more up to date than the pdf i talked about.
The trend suggests that the number of v4 users are dropping fast.
So maybe there could be a v4 fallback for now till we are convinced that there are very few of v4 users left before deprecating it

**Comment by GochoMugo** [comment]
So, I think we should push our community to using newer versions of Node.js. So we could deprecate older versions continuously. So it seems we will continue transpiling until the moment we will deprecate Node v4.x!

**Comment by crazyabdul** [comment]
@gochoMugo should i go ahead and make a pull request?

**Comment by preco21** [comment]
@crazyabdul Good to go.

IMO, transpiles are needed. I don't think it makes sense to simply change the codebase to the old grammar to support older version of the node.

Newer grammar gives us readability, maintainable code, such advantages.

**Comment by GochoMugo** [comment]
@crazyabdul Everything seems okay. We are waiting for a PR to fallback Node v4.x to use transpiled code. (@preco21's [comment](https://github.com/yagop/node-telegram-bot-api/issues/275#issuecomment-277676542) above is helpful.)

@preco21 I agree. At the rate at which Node releases newer versions, it seems unwise to downgrade our syntax to support older versions, which will be surpassed in quite a short time I presume.

**Comment by crazyabdul** [comment]
@GochoMugo Pull request (https://github.com/yagop/node-telegram-bot-api/pull/280) sent

**Comment by preco21** [comment]
Great job guys!

**Comment by GochoMugo** [comment]
PR #280 merged into master in commit [d4a469df6b6eeae4c045faf7b6e63bdc765b0a15](https://github.com/yagop/node-telegram-bot-api/commit/d4a469df6b6eeae4c045faf7b6e63bdc765b0a15). Please try out the fix and report your results.

**Comment by crazyabdul** [comment]
@GochoMugo @preco21 @yagop it's working now. I was able to run `node index.js` with no problems on v4.4.3. Didn't have to add the `--use_strict` option. 

Thanks for letting me help out. Deeply appreciated

**Comment by GochoMugo** [comment]
Fixed in **v0.27.0**.
